phpstan.neon.dist

```diff
@@ -106,5 +106,11 @@ parameters:
             message: '~^Instanceof between Doctrine\\DBAL\\Platforms\\Keywords\\KeywordList and Doctrine\\DBAL\\Platforms\\Keywords\\KeywordList will always evaluate to true\.~'
             paths:
                 - %currentWorkingDirectory%/src/Platforms/AbstractPlatform.php
+
+        # We're checking for invalid invalid input
```

Comment:
It looks like this error shouldn't be manually suppressed given that we disable [`treatPhpDocTypesAsCertain`](https://phpstan.org/config-reference#treatphpdoctypesascertain): https://github.com/doctrine/dbal/blob/256da038ef3e5bb33a35e93520680661ef542f01/phpstan.neon.dist#L5

Is it really necessary?

Reply:
Yes PHPStan complained about it, which is why I added this entry. IIRC, PHPStan would also complain, if that ignore rule didn't match any errors. If you have a better idea how I could make PHPStan happy here, I'm all ears. 🙂

Reply:
> IIRC, PHPStan would also complain, if that ignore rule didn't match any errors.

Not in our setup: https://github.com/doctrine/dbal/blob/aefa1a97b274265b901ee8499c53387c10867700/phpstan.neon.dist#L6

> If you have a better idea how I could make PHPStan happy here, I'm all ears.

It's not only about making it happy. Usually, the process is that each suppressed issue annotatied either with a link to existing issue in the DBAL or an issue reported at the PHPStan bug tracker. This way, there is an understanding of how to proceed with getting rid of this suppression.

This case looks like a bug that I cannot [reproduce](https://phpstan.org/r/48ba4d9e-05fb-4c82-8f67-ea8e0954b0e1) in their playground: the error is only reproducible when "Treat PHPDoc types as certain" is checked but in our case it's not.

Reply:
Given that this is a temporary code, we can leave it as is. I'll try to get to the bottom of the issue asynchronously.